packages/ui/src/components/pressable/pressable.native.tsx

```diff
@@ -0,0 +1,55 @@
+import { ComponentPropsWithoutRef } from 'react';
+import { Pressable as RNPressable, type PressableProps as RNPressableProps } from 'react-native';
+import Animated from 'react-native-reanimated';
+
+import {
+  BackgroundColorShorthandProps,
+  BorderProps,
+  LayoutProps,
+  OpacityProps,
+  PositionProps,
+  SpacingShorthandProps,
+  VisibleProps,
+  backgroundColorShorthand,
+  border,
+  createRestyleComponent,
+  layout,
+  opacity,
+  position,
+  spacingShorthand,
+  visible,
+} from '@shopify/restyle';
+
+import { Theme } from '../../theme-native';
+
+export type PressableBaseProps = RNPressableProps &
+  OpacityProps<Theme> &
+  VisibleProps<Theme> &
+  SpacingShorthandProps<Theme> &
+  BorderProps<Theme> &
+  BackgroundColorShorthandProps<Theme> &
+  LayoutProps<Theme> &
+  PositionProps<Theme>;
+
+export const buttonRestyleFunctions = [
+  opacity,
+  visible,
+  spacingShorthand,
+  border,
+  backgroundColorShorthand,
+  layout,
+  position,
+];
+
+const PressableBase = createRestyleComponent<PressableBaseProps, Theme>(
+  buttonRestyleFunctions,
+  RNPressable
+);
+
+export const Pressable = Animated.createAnimatedComponent(PressableBase);
+export interface PressableProps extends ComponentPropsWithoutRef<typeof Pressable> {
+  // Incorrect typings in: https://github.com/software-mansion/react-native-reanimated/blob/main/packages/react-native-reanimated/src/helperTypes.ts#L48
+  // resulting in inferred key prop of createAnimatedComponent(X) to be incompatible with itself.
+  // TODO: Needs further investigation/raising an issue.
+  key?: any;
```

Comment:
Do you get an error if you don't have this key set to `any`?

Reply:
A very nasty one, yes. This is a typings issue stemming from react-native-reanimated that isn't straightforward (if at all possible) to fix in userland. It's low priority, but in my list try to (reluctantly 😂) take another look.

Reply:
OK, that's fair enough to leave it as so 👍  thanks!